tests/unittest_brain.py

```diff
@@ -3126,5 +3126,50 @@ def __len__(self) -> int:
         node.inferred()
 
 
+def test_inference_on_outer_referential_length_check() -> None:
+    """
+    Regression test for https://github.com/PyCQA/pylint/issues/5244
+    """
```

Comment:
Can you add a small note here that the inference should succeed? Might make it a bit more obvious.